### PR TITLE
Make book restful

### PR DESCRIPTION
**Pull request by padu12**

### CODE REVIEW

**Review by boroda4436 — COMMENTED** [review]
В тебе в цьому PR дублюється код з попереднього пул реквеста. Спочатку треба отримати аппрув на попередній PR, замерджити його в мастер, а потім відкривати ПР зі змінами які стосуються конкретно цієї задачі

File: src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java]: https://medium.com/@centizennationwide/mysql-naming-conventions-e3a6f6219efe use singular work for table discription

File: src/main/java/com/kaziamyr/onlinebookstore/dto/BookDto.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import lombok.Data;
+
+@Data
+public class BookDto {
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private String price;
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/BookDto.java]: Have you tested your app using Postman? If so, then you were all right with the price string type? I think it's wrong, you should use BigDecimal

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.kaziamyr.onlinebookstore.dto;
+
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
+    private String title;
+    private String author;
+    private String isbn;
+    private String price;
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java]: the same

File: src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -47,7 +48,16 @@ public List<Book> findAll() {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery(query, Book.class).getResultList();
         } catch (Exception e) {
-            throw new HibernateException("Can't get all books from db");
+            throw new EntityNotFoundException("Can't get all books from db", e);
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java:47]:
```
throw new EntityNotFoundException("Can't get all books", e);
```

File: src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package com.kaziamyr.onlinebookstore.repository.impl;
 
+import com.kaziamyr.onlinebookstore.exception.EntityNotFoundException;
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java:3]: You don't need the Autowired annotation, spring understands it automatically. Anyway it is better to use the annotation that we have studied - @RequiredArgsConstructor

File: src/main/java/com/kaziamyr/onlinebookstore/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -10,19 +13,29 @@
...
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
     @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
         this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
     }
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/service/impl/BookServiceImpl.java]: the same

**Review by Den-moroz — CHANGES_REQUESTED** [review]
Good job) Let's improve your solution

File: src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java (hunk 1 above)
Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/controller/BookController.java]: It is better to make /api/books

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.kaziamyr.onlinebookstore.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Data
+public class CreateBookRequestDto {
```
Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:7]: Do you really need this class, maybe it is possibe to reuse BookDto they have almost all same fields

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some improvements required

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java (hunk 7 above)
Comment (Vasul-Pankiv) [reply src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:7]: I also think that you could reuse BookDto, but its up to you

**Review by Vasul-Pankiv — APPROVED** [review]

File: src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java (hunk 7 above)
Comment (padu12) [reply src/main/java/com/kaziamyr/onlinebookstore/dto/CreateBookRequestDto.java:7]: https://mateacademy.slack.com/archives/C0554BPUYJ1/p1692708245197979

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**